src/i_oalmusic.c

```diff
@@ -0,0 +1,314 @@
+//
+// Copyright(C) 2023 Roman Fomin
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//
+
+#include "i_oalmusic.h"
+
+#include <AL/al.h>
+#include <AL/alext.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "SDL.h"
+
+#include "doomtype.h"
+#include "i_sndfile.h"
+#include "i_sound.h"
+
+// Define the number of buffers and buffer size (in milliseconds) to use. 4
```

Comment:
I don't quite understand why we need 4 buffers for streaming music?

Reply:
OpenAl wants to queue buffers to ensure smooth playback. See `alSourceQueueBuffers`